bigtable/client/table.cc

```diff
@@ -133,6 +133,47 @@ void Table::BulkApply(BulkMutation&& mut) {
   }
 }
 
+// Call the `google.bigtable.v2.Bigtable.SampleRowKeys` RPC until
+// successful. When RPC is finished, this function returns the SampleRowKeys
+// as a Collection specified by the user. If the RPC fails, it will keep
+// retrying until the policies in effect tell us to stop.
+void Table::sample_rows_impl(std::function<void(Table::RowKeySample)> inserter,
+                             std::function<void()> clearer) {
+  // Copy the policies in effect for this operation.
+  auto backoff_policy = rpc_backoff_policy_->clone();
+  auto retry_policy = rpc_retry_policy_->clone();
+
+  // Build the RPC request for SampleRowKeys
+  btproto::SampleRowKeysRequest request;
+  btproto::SampleRowKeysResponse response;
+  request.set_table_name(table_name_);
+
+  while (true) {
+    grpc::ClientContext client_context;
+    backoff_policy->setup(client_context);
+    retry_policy->setup(client_context);
+
+    auto stream = client_->Stub()->SampleRowKeys(&client_context, request);
+    while (stream->Read(&response)) {
+      // Assuming collection will be either list or vector.
+      Table::RowKeySample row_sample;
+      row_sample.offset_bytes = response.offset_bytes();
+      row_sample.row_key = response.row_key();
```

Comment:
can we move the `row_key()` contents from `response` into `row_sample` to avoid copying the string?